modules/041-linstor/templates/nodegroupconfiguration-drbd-install-altlinux-like.yaml

```diff
@@ -0,0 +1,127 @@
+    {{- else }}
+    need_drbd="true"
+    {{- end }}
+
+    bb-log-info "we need drbd on node: "$need_drbd
+
+    if [ $need_drbd == "true" ]; then
+      CLUSTER_DNS="{{ .Values.global.discovery.clusterDNSAddress }}"
+      SPAAS_FQDN="spaas.d8-linstor.svc.{{ .Values.global.discovery.clusterDomain }}"
+
+      if [ -e "/proc/drbd" ]; then
+        # DRBD check version
+
+        current_version="$(cat /proc/drbd | grep 'version:' | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')"
+        desired_version="{{ $.Values.linstor.internal.drbdVersion }}"
+
+        # We expect the loaded DRBD module to be version 9.
+        # If version 8 is loaded, it means that for some reason, the in-tree kernel module has been automatically loaded.
+        # (For example, this can happen due to drbd-utils installed on the host, which should not occur in standard scenarios).
+        # We are only interested in the version 9 loaded by our helper script, so unload module and wait until it done.
+        if [[ ! $current_version =~ ^9.* ]]; then
+          rmmod drbd_transport_rdma || true
+          rmmod drbd_transport_tcp || true
+          rmmod drbd || true
+        fi
+
+        if [ "${current_version}" == "${desired_version}" ]; then
+          bb-log-info "Actual drbd version loaded"
+          exit 0
+        else
+          bb-log-info "Non-actual version of drbd (now "$current_version", desired "$desired_version"), setting reboot flag"
+          bb-flag-set reboot
+        fi
+      fi
+
+      bb-rp-install "drbd924:{{ .Values.global.modulesImages.digests.registrypackages.drbd924 }}"
+      # Here we install headers for further usage. It can be different from current kernel version! So further we check, if we need new kernel version.
+      apt-get -y install make gcc bind-utils patch kernel-headers-modules-std-def
+
+      attempt=0
+      until [ `host -t A "$SPAAS_FQDN" "$CLUSTER_DNS" | grep -q 'has address' && echo 1 || echo 0` == 1 ]
+      do
+        if [ $attempt -gt 60 ]; then
+          bb-log-info "No answer from DNS"
+          exit 1
+        fi
+
+        ((attempt=attempt+1))
+        bb-log-info "Waiting for DNS response (try #$attempt)"
+        sleep 10
+      done
+
+      bb-log-info "DNS responded, continue DRBD build"
+
+      SPAAS_IP="$(host -t A "$SPAAS_FQDN" "$CLUSTER_DNS" | awk '/has address/ { print $4 ; exit }')"
+      SPAAS_URL="https://${SPAAS_IP}:2020"
+
+      attempt=0
+      until [ `d8-curl -kIs $SPAAS_URL'/api/v1/hello' | grep -q 'HTTP/2 405' && echo 1 || echo 0` == 1 ]
+      do
+        if [ $attempt -gt 60 ]; then
+          bb-log-info "No answer from SPAAS"
+          exit 1
+        fi
+
+        ((attempt=attempt+1))
+        bb-log-info "Waiting for SPAAS answer (try #$attempt)"
+        sleep 10
+      done
+
+      bb-log-info "SPAAS responded, continue DRBD build"
+
+      cd /opt/deckhouse/drbd
+      need_kernel_update="false"
+      # If there is no header - we need updated kernel
```

Comment:
We need a way to detect missing headers without starting make.
make could fail with other reasons.